phpBB/phpbb/ucp/controller/webpush.php

```diff
@@ -178,23 +196,55 @@ private function get_anonymous_notifications(): string
 			$push_token = $notification_row['push_token'];
 
 			// Check if passed push token is valid
-			$sql = 'SELECT user_form_salt
+			$sql = 'SELECT user_form_salt, user_lang
 				FROM ' . USERS_TABLE . '
 				WHERE user_id = ' . (int) $user_id;
 			$result = $this->db->sql_query($sql);
-			$user_form_token = $this->db->sql_fetchfield('user_form_salt');
+			$row = $this->db->sql_fetchrow($result);
 			$this->db->sql_freeresult($result);
 
+			$user_form_token = $row['user_form_salt'];
+			$user_lang = $row['user_lang'];
+
 			$expected_push_token = hash('sha256', $user_form_token . $push_token);
 			if ($expected_push_token === $token)
 			{
-				return $notification_data;
+				if ($user_lang !== $this->language->get_used_language())
+				{
+					$this->language->set_user_language($user_lang, true);
+				}
+				return $this->get_notification_data($notification_data);
 			}
 		}
 
 		throw new http_exception(Response::HTTP_FORBIDDEN, 'NO_AUTH_OPERATION');
 	}
 
+	private function get_notification_data(string $notification_data): string
+	{
+		$row_data = json_decode($notification_data, true);
+
+		// Old notification data is pre-parsed and just needs to be returned
+		if (isset($row_data['heading']))
+		{
+			return $notification_data;
+		}
+
+		// Get notification from row_data
+		$notification = $this->notification_manager->get_item_type_class($row_data['notification_type_name'], $row_data);
+
+		// Load users for notification
+		$this->user_loader->load_users($notification->users_to_query());
```

Comment:
Should you add `[USER_IGNORE]` as 2nd parameter here?

Reply:
I don't think that's needed and also not what is done in the board notifications.

Reply:
I was just asking because we use this same method in the other webpush class (the method class) and there we are using the 2nd parameter with USER_IGNORE

Reply:
Though that is before creating the notification. Here it's about displaying it correctly so there shouldn't be any of these left.